src/util/psbt/map/output.rs

```diff
@@ -92,33 +92,47 @@ impl PartialEq for TapTree {
 
 impl Eq for TapTree {}
 
+impl From<TapTree> for TaprootBuilder {
+    #[inline]
+    fn from(tree: TapTree) -> Self {
+        tree.into_builder()
+    }
+}
+
 impl TapTree {
     /// Gets the inner node info as the builder is finalized.
-    fn node_info(&self) -> &NodeInfo {
+    pub fn node_info(&self) -> &NodeInfo {
         // The builder algorithm invariant guarantees that is_complete builder
         // have only 1 element in branch and that is not None.
         // We make sure that we only allow is_complete builders via the from_inner
         // constructor
         self.0.branch()[0].as_ref().expect("from_inner only parses is_complete builders")
     }
 
-    /// Converts a [`TaprootBuilder`] into a tree if it is complete binary tree.
+    /// Constructs [`TapTree`] from a [`TaprootBuilder`] if it is complete binary tree.
     ///
     /// # Return
-    /// A `TapTree` iff the `inner` builder is complete, otherwise return the inner as `Err`.
-    pub fn from_inner(inner: TaprootBuilder) -> Result<Self, TaprootBuilder> {
-        if inner.is_complete() {
-            Ok(TapTree(inner))
+    /// A `TapTree` iff the `builder` builder is complete, otherwise return the inner as `Err`.
+    pub fn from_builder(builder: TaprootBuilder) -> Result<Self, TaprootBuilder> {
+        if builder.is_complete() {
+            Ok(TapTree(builder))
         } else {
-            Err(inner)
+            Err(builder)
         }
     }
 
     /// Converts self into builder [`TaprootBuilder`]. The builder is guaranteed to be finalized.
-    pub fn into_inner(self) -> TaprootBuilder {
+    pub fn into_builder(self) -> TaprootBuilder {
         self.0
     }
 
+    /// Returns reference for the [`TaprootBuilder`] stored internally. The builder is guaranteed
+    /// to be finalized.
+    #[inline]
+    pub fn as_builder(&self) -> &TaprootBuilder {
+        &self.0
+    }
```

Comment:
You forgot to remove this one which also leaks internal representation.

Reply:
It can't be used for modifying the tree structure and is required to avoid a lot of clones (like `tap_tree.clone().into_builder()` downstream.

Reply:
Does it cause significant performance issues? Perhaps there's a more specific pattern of operations you often perform and could be handled differently?

Reply:
Actually, looking at `TaprootBuilder` API it doesn't accept `&self` in any modifying method so you'd still have to write `tap_tree.as_builder().clone()` and there's no perf difference. If the issue is just annoyance, I'm open to:

```rust
pub fn to_builder(&self) -> TaprootBuilder {
    self.clone().into_builder()
}
```

Reply:
Did that